recipes/installer.rb

```diff
@@ -25,7 +25,9 @@
   action node['cwlogs']['attempt_upgrade'] ? :create : :create_if_missing
 end
 
+proxy_env = ENV.select { |k, v| %w(http_proxy https_proxy no_proxy).include?(k) && !v.empty? }
+proxy_args = proxy_env.map { |k,v| "--#{k.gsub('_','-')} '#{v}'" }.join(' ')
 execute 'Install CloudWatch Logs agent' do
-  command "/opt/aws/cloudwatch/awslogs-agent-setup.py -n -r #{node['cwlogs']['region']} -c /tmp/cwlogs.cfg"
-  not_if { system 'pgrep -f aws-logs-agent-setup' }
+  command "/opt/aws/cloudwatch/awslogs-agent-setup.py -n -r #{node['cwlogs']['region']} -c /tmp/cwlogs.cfg #{proxy_args}"
+  not_if "pgrep -af awslogs-agent-setup"
```

Comment:
- `-a` is not a valid option of the pgrep in  Amazon linux (and I guess most other Linux OS'es). This would need an os detection or stay without `-a` to be merged.
- We should be using single quotes here